server/game/cards/03_TWI/units/TheInvisibleHandImposingFlagship.ts

```diff
@@ -0,0 +1,40 @@
+import AbilityHelper from '../../../AbilityHelper';
+import { NonLeaderUnitCard } from '../../../core/card/NonLeaderUnitCard';
+import { RelativePlayer, TargetMode, Trait, WildcardCardType, WildcardZoneName } from '../../../core/Constants';
+
+export default class TheInvisibleHandImposingFlagship extends NonLeaderUnitCard {
+    protected override getImplementationId () {
+        return {
+            id: '0398102006',
+            internalName: 'the-invisible-hand#imposing-flagship',
+        };
+    }
+
+    public override setupCardAbilities(): void {
+        this.addWhenPlayedAbility({
+            title: 'Create 4 Battle Droid tokens.',
+            immediateEffect: AbilityHelper.immediateEffects.createBattleDroid({ amount: 4 }),
+        });
+
+        this.addOnAttackAbility({
+            title: 'Exhaust any number of friendly Separatist units',
+            optional: true,
+            targetResolver: {
+                mode: TargetMode.Unlimited,
+                controller: RelativePlayer.Self,
+                cardTypeFilter: WildcardCardType.Unit,
+                zoneFilter: WildcardZoneName.AnyArena,
+                cardCondition: (card) => card.hasSomeTrait(Trait.Separatist) && card.isPlayable() && !card.exhausted,
```

Comment:
My first test with the tokens does not work, because I do not think tokens are playable.
But without the `isPlayable` test and I can't call the `exhausted` property.
There must be something obvious I am missing, but I am stuck on this card because implementation because of this `exhausted` that I can test easily.